cmake/FindCVC4.cmake

```diff
@@ -1,8 +1,26 @@
 if (USE_CVC4)
     find_path(CVC4_INCLUDE_DIR cvc4/cvc4.h)
-    find_library(CVC4_LIBRARY NAMES cvc4 )
+    find_library(CVC4_LIBRARY NAMES cvc4)
     include(FindPackageHandleStandardArgs)
     find_package_handle_standard_args(CVC4 DEFAULT_MSG CVC4_LIBRARY CVC4_INCLUDE_DIR)
+    if(CVC4_FOUND)
+        # CVC4 may depend on either CLN or GMP.
+        # We can assume that the one it requires is present on the system,
+        # so we quietly try to find both and link against them, if they are
+        # present.
```

Comment:
I'm fine with this, but likely we may run into a compilation issue reported by someone where this isn't the case. Think about statically linked CVC4 against GMP?

Reply:
Discussed this on gitter. Resolution: 
> Perhaps we should include that in the changelog: should you encounter issues with the experimental z3 / cvc4 support, please disable them via `USE_Z3=OFF USE_CVC4=OFF` when configuring cmake.

Reply:
Added some Changelog entry (I'm open to suggestions).